### DISCUSSION TITLE
Weird failed migration behavior

### DISCUSSION THREAD

**Discussion by vuki656** [mikro-orm/mikro-orm#5430] [General]
Hey,

Im opening a discussion since in not sure if this is a bug or just misconfiguration/misunderstanding on my part. 

The problem I encountered is if in a migration we do 2 things, say create a table and then run a raw query, if the raw query fails, the table creation won't be reverted in the database. 

![img](https://github.com/mikro-orm/mikro-orm/assets/8104698/5b48299f-dac8-4405-916b-5a798831663a)

![image](https://github.com/mikro-orm/mikro-orm/assets/8104698/a6b54ee5-aa20-49bc-be20-47701ba95b2b)

Here is a repro repo: https://github.com/vuki656/micro-orm-migration-repro

If wrap the migration up in a knex transaction, it will get reverted correctly. 

Thanks for the great lib!

**Comment by Spolja** [comment]
Bumping this one as its really annoying not to have this by default.

**Comment by B4nan** [comment]
You should not run queries like that, you are completely ignoring the transaction context here. Use `this.execute()` instead, which accepts both SQL strings as well as knex QB instances. Alternatively, there is `this.ctx` which is the transaction context (basically a knex connection which you can feed into `qb.transactional` on the knex side). `this.getKnex()` does not hold transaction context, it can not - it is an alternative to doing `knex()`, its creating the knex object, no transactions are involved in there.

**Reply by vuki656** [reply]
Thank you for the clarification
